packages/api/internal/handlers/sandbox_get.go

```diff
@@ -0,0 +1,72 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/e2b-dev/infra/packages/api/internal/api"
+	"github.com/e2b-dev/infra/packages/api/internal/auth"
+	authcache "github.com/e2b-dev/infra/packages/api/internal/cache/auth"
+	"github.com/e2b-dev/infra/packages/shared/pkg/models/envbuild"
+	"github.com/e2b-dev/infra/packages/shared/pkg/telemetry"
+)
+
+func (a *APIStore) GetSandboxesSandboxID(c *gin.Context, id string) {
+	ctx := c.Request.Context()
+
+	teamInfo := c.Value(auth.TeamContextKey).(authcache.AuthTeamInfo)
+	team := teamInfo.Team
+
+	telemetry.ReportEvent(ctx, "get running instance")
+
+	info, err := a.instanceCache.GetInstance(id)
+	if err != nil {
+		c.String(http.StatusNotFound, fmt.Sprintf("instance \"%s\" doesn't exist or you don't have access to it", id))
+		return
+	}
+
+	if *info.TeamID != team.ID {
+		c.JSON(http.StatusNotFound, fmt.Sprintf("instance \"%s\" doesn't exist or you don't have access to it", id))
+		return
+	}
+
+	a.posthog.IdentifyAnalyticsTeam(team.ID.String(), team.Name)
+	properties := a.posthog.GetPackageToPosthogProperties(&c.Request.Header)
+	a.posthog.CreateAnalyticsTeamEvent(team.ID.String(), "get running instance", properties)
+
+	build, err := a.db.Client.EnvBuild.Query().Where(envbuild.ID(*info.BuildID)).First(ctx)
+	if err != nil {
+		telemetry.ReportCriticalError(ctx, err)
+		c.JSON(http.StatusInternalServerError, nil)
```

Comment:
Could also add a message here
```suggestion
		c.JSON(http.StatusInternalServerError, nil)
```